internal/controllers/configuration/object_references.go

```diff
@@ -0,0 +1,165 @@
+}
+
+func listNetV1IngressReferredSecrets(ingress *netv1.Ingress) []types.NamespacedName {
+	referredSecretNames := make([]types.NamespacedName, 0, len(ingress.Spec.TLS))
+	for _, tls := range ingress.Spec.TLS {
+		nsName := types.NamespacedName{
+			Namespace: ingress.Namespace,
+			Name:      tls.SecretName,
+		}
+		referredSecretNames = append(referredSecretNames, nsName)
+	}
+
+	return referredSecretNames
+}
+
+func listNetV1beta1IngressReferredSecrets(ingress *netv1beta1.Ingress) []types.NamespacedName {
+	referredSecretNames := make([]types.NamespacedName, 0, len(ingress.Spec.TLS))
+	for _, tls := range ingress.Spec.TLS {
+		nsName := types.NamespacedName{
+			Namespace: ingress.Namespace,
+			Name:      tls.SecretName,
+		}
+		referredSecretNames = append(referredSecretNames, nsName)
+	}
+
+	return referredSecretNames
+}
+
+func listExtensionV1beta1IngressReferredSecrets(ingress *extv1beta1.Ingress) []types.NamespacedName {
+	referredSecretNames := make([]types.NamespacedName, 0, len(ingress.Spec.TLS))
+
+	for _, tls := range ingress.Spec.TLS {
+		nsName := types.NamespacedName{
+			Namespace: ingress.Namespace,
+			Name:      tls.SecretName,
+		}
+		referredSecretNames = append(referredSecretNames, nsName)
+	}
+
+	return referredSecretNames
+}
+
+func listKongPluginReferredSecrets(plugin *kongv1.KongPlugin) []types.NamespacedName {
+	referredSecretNames := make([]types.NamespacedName, 0, 1)
+
+	if plugin.ConfigFrom != nil {
+		nsName := types.NamespacedName{
+			Namespace: plugin.Namespace,
+			Name:      plugin.ConfigFrom.SecretValue.Secret,
+		}
+		referredSecretNames = append(referredSecretNames, nsName)
+	}
+
+	return referredSecretNames
+}
+
+func listKongClusterPluginReferredSecrets(plugin *kongv1.KongClusterPlugin) []types.NamespacedName {
+	referredSecretNames := make([]types.NamespacedName, 0, 1)
+
+	if plugin.ConfigFrom != nil {
+		nsName := types.NamespacedName{
+			Namespace: plugin.ConfigFrom.SecretValue.Namespace,
+			Name:      plugin.ConfigFrom.SecretValue.Secret,
+		}
+		referredSecretNames = append(referredSecretNames, nsName)
+	}
+
+	return referredSecretNames
+}
+
+func listKongConsumerReferredSecrets(consumer *kongv1.KongConsumer) []types.NamespacedName {
+	referredSecretNames := make([]types.NamespacedName, 0, len(consumer.Credentials))
+
+	for _, secretName := range consumer.Credentials {
+		nsName := types.NamespacedName{
+			Namespace: consumer.Namespace,
+			Name:      secretName,
+		}
+		referredSecretNames = append(referredSecretNames, nsName)
+	}
+
+	return referredSecretNames
+}
+
+func listTCPIngressReferredSecrets(tcpIngress *kongv1beta1.TCPIngress) []types.NamespacedName {
+	referredSecretNames := make([]types.NamespacedName, 0, len(tcpIngress.Spec.TLS))
+	for _, tls := range tcpIngress.Spec.TLS {
+		nsName := types.NamespacedName{
+			Namespace: tcpIngress.Namespace,
+			Name:      tls.SecretName,
+		}
+		referredSecretNames = append(referredSecretNames, nsName)
+	}
+
+	return referredSecretNames
+}
```

Comment:
There is no better way? 😱

Reply:
What do you mean by a "better" way? Referred secrets lies in different place in different kind of k8s object. I do not believe there will be a simple function that magically returns all secrets referred (without endless `switch-case` followed by the processing code).